types/coin.go

```diff
@@ -19,6 +19,11 @@ func (coin Coin) String() string {
 	return fmt.Sprintf("%v%v", coin.Amount, coin.Denom)
 }
 
+// SameDenomAs returns true if the two coins are the same denom
+func (coin Coin) SameDenomAs(other Coin) bool {
+	return (coin.Denom == other.Denom)
```

Comment:
do we really need this as a function? It seems obvious just to check this manually when you want.